server/plugins/ML_SpecStats.cpp

```diff
@@ -209,7 +209,10 @@ void SpecPcile_next(SpecPcile* unit, int inNumSamples) {
         // Used to be MAKE_TEMP_BUF but we can handle it more cleanly in this specific case:
         if (!unit->m_tempbuf) {
         unit->m_tempbuf = (float*)RTAlloc(unit->mWorld, numbins * sizeof(float));
-        ClearUnitIfMemFailed(unit->m_tempbuf);
+        if (!unit->m_tempbuf) {
+            ClearUnitOutputs(unit, inNumSamples);
+            ClearUnitOnMemFailed;
```

Comment:
The example plugins also show that they "map" the next_calc function to `ft->fClearUnitOutputs`, see https://github.com/supercollider/example-plugins/blob/d0622ac6a4919bc7db134561d292d71337a9823d/03-AnalogEcho/AnalogEcho.cpp#L47-L53

Is this relevant here? Don't have too much experience with rt malloc (yet).

Reply:
Yes, it's done by `ClearUnitOnMemFailed`:
```
#define ClearUnitOnMemFailed                                                                                           \
    Print("%s: alloc failed, increase server's RT memory (e.g. via ServerOptions)\n", __func__);                       \
    SETCALC(*ClearUnitOutputs);                                                                                        \
    unit->mDone = true;                                                                                                \
    return;
```